File: c/sum-of-multiples/sum_of_multiples.c
#include "sum_of_multiples.h"

unsigned int sum( const unsigned int *factors, const size_t number_of_factors,
        const unsigned int limit) {
    unsigned int sum = 0;
    for (unsigned int n = 1; n < limit; ++n) {
        for (size_t i = 0; i < number_of_factors; ++i) {
            if (factors[i] == 0) continue;
            if (n % factors[i] == 0) {
                sum += n;
                break;
            }
        }
    }
    return sum;
}

File: zig/sum-of-multiples/sum_of_multiples.zig
pub fn sum(_: anytype, factors: []const u32, limit: u32) !u64 {
    var result: u64 = 0;
    var n: u64 = 1;
    while (n < limit) {
        for (factors) |f| {
            if (f != 0 and n % f == 0) {
                result += n;
                break;
            }
        }
        n += 1;
    }
    return result;
}
